.github/workflows/tox.yml

```diff
@@ -34,6 +34,12 @@ jobs:
     - name: Install python dependencies
       run: |
         python -m pip install --upgrade pip setuptools wheel
+    - name: Set up C++ devtools
+      uses: seanmiddleditch/gha-setup-ninja@v3
```

Comment:
Why not use the linux default, i.e. `Unix Makefiles`? It's more likely to be on a system than ninja.

Reply:
Make is not on every system, not even every Linux system by default. If you have to tell users to install Make, you might as well just tell them to install Ninja. Legend says Ninja is faster.

Reply:
I disagree, I doubt you will find a linux system with a c++ compiler that doesn't have make, but you will find plenty that don't have ninja. Build system speed here is irrelevant because we compile a single file. I will not require to change it, but I don't see a reason to deviate from the default.

Reply:
I don't want CMake to pick something automatically because then I have to figure out if it's a multi-config or single-config generator and change the CMakeFile.txt and the build commands accordingly. Ninja is the only decent tool that works everywhere. Ninja is also faster for incremental builds and no-op builds, which may or may not be relevant depending on how we do the caching.
These are not really killer features, and they may not be used at the moment, but it may save some headache in the future, especially if people add to it while following the example of ignoring these corner cases.
Any differences on the user's end should anyways disappear once we get the library dependencies and build tools through spack.

Reply:
>Any differences on the user's end should anyways disappear once we get the library dependencies and build tools through spack.

I don't understand the link to spack here. I don't think we want to require spack from GT4Py users.

Reply:
Maybe worth mentioning that `ninja` is already a build-time default for `pybind11`, one of our dependencies. From there we could reason that on any platform where the `pybind11` pypi is available, the `ninja` one should be too.

Reply:
It now uses cmake and ninja from PyPI.